homeassistant/components/climate/isy994.py

```diff
@@ -0,0 +1,321 @@
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if self._temp_unit:
+            return self._temp_unit
+        return self.hass.config.units.temperature_unit
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._current_humidity
+
+    @property
+    def current_operation(self) -> str:
+        """Return current operation ie. heat, cool, idle."""
+        return self._current_operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [STATE_HEAT,
+                STATE_COOL,
+                STATE_AUTO,
+                STATE_PROG_AUTO,
+                STATE_FAN_ONLY,
+                STATE_OFF]
+        # STATE_PROG_COOL & STATE_PROG_HEAT are valid possible states,
+        # but can't be set by the ISY via REST interface for some reason.
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.value
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    @property
+    def target_temperature_high(self):
+        """Return the highbound target temperature we try to reach."""
+        return self._target_temp_high
+
+    @property
+    def target_temperature_low(self):
+        """Return the lowbound target temperature we try to reach."""
+        return self._target_temp_low
+
+    @property
+    def is_on(self):
+        """Return if the unit is on or off based on operation status."""
+        if self._current_status not in [STATE_OFF, STATE_IDLE, STATE_UNKNOWN]:
+            return True
+        return False
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        return [STATE_AUTO, STATE_ON]
+
+    @property
+    def current_fan_mode(self) -> str:
+        """Return the current fan mode ie. auto, on."""
+        return self._fan_mode
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return (SUPPORT_OPERATION_MODE |
+                SUPPORT_FAN_MODE |
+                SUPPORT_TARGET_TEMPERATURE_HIGH |
+                SUPPORT_TARGET_TEMPERATURE_LOW)
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(DEFAULT_MIN_TEMP, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return convert_temperature(DEFAULT_MAX_TEMP, TEMP_CELSIUS,
+                                   self.temperature_unit)
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        data = {
+            ATTR_CURRENT_TEMPERATURE: self.current_temperature,
```

Comment:
Don't list all these things again, they are already handled by the climate class.

Reply:
I attempted to remove this and got several errors because when `device_state_attributes` is called it falls back to the definition in `isy994/__init__.py` which doesn't handle the response correctly.